docs/code-standards-policy.rst

```diff
@@ -118,36 +171,418 @@ using the debops.apt_install_ role MUST be defined as (or similar):
+``--verbose`` options. For example:
+
+.. code-block:: yaml
+
+   - name: Show intermediate value from a lookup query
+     debug:
+       var: '{{ lookup("template", "lookups/fancy_lookup.j2") }}'
+       verbosity: 1
+
+
+.. _debops_policy__ref_code_standards_role_dependencies:
+
+Ansible role dependencies
+-------------------------
+
+.. _debops_policy__ref_code_standards_soft_role_dependencies:
+
+Soft role dependencies
+~~~~~~~~~~~~~~~~~~~~~~
+
+Role dependencies are considered "soft" if they are defined in the ``roles``
+list of a playbook. This approach offers a higher flexibility as the user can
+choose which playbooks to run and which features to include.
+
+Whenever possible DebOps role authors MUST specify role dependencies via
+playbook instead of
+:ref:`"hard" dependencies <debops_policy__ref_code_standards_hard_role_dependencies>`
+in the :file:`meta/main.yml`.
+
+It's also possible to pass configuration values to a dependent role if the
+involved role is offering dedicated
+:ref:`dependent variables <debops_policy__ref_code_standards_dependent_variables>`
+for this purpose. With soft dependencies custom playbook authors are therefore
+free to pass values from arbitrary sources according to their requirements.
+
+**Example:**
+
+This is an example playbook for debops.slapd_ defining soft dependencies:
+
+.. code-block:: yaml
+
+    ---
+
+    - name: Manage OpenLDAP service
+      hosts: [ 'debops_service_slapd', 'debops_slapd' ]
+      become: True
+
+      roles:
+
+        - role: debops.ferm
+          tags: [ 'role::ferm' ]
+          ferm__dependent_rules:
+            - '{{ slapd_ferm_dependent_rules }}'
+
+        - role: debops.tcpwrappers
+          tags: [ 'role::tcpwrappers' ]
+          tcpwrappers_dependent_allow:
+            - '{{ slapd_tcpwrappers_dependent_allow }}'
+
+        - role: debops.slapd
+          tags: [ 'role::slapd' ]
+
+
+.. _debops_policy__ref_code_standards_hard_role_dependencies:
+
+Hard role dependencies
+~~~~~~~~~~~~~~~~~~~~~~
+
+Role dependencies are considered "hard" if they are defined in the
+``dependencies`` list in :file:`meta/main.yml`.  DebOps role authors MUST
+avoid the use of hard role dependencies for the follwing reasons:
+
+- Hard role dependencies must always be installed on the Ansible controller
+  even when their execution is conditionally triggered via ``when`` statement.
+
+- It hinders the independent use of the role in a custom playbook or outside
+  of DebOps where playbook authors might relay on a different role for a certain
+  feature or decide no to use a certain feature at all.
+
+- The playbook execution flow is more difficult to reason about as hard
+  dependencies are defined outside of the playbook.
+
+Generally role dependencies MUST be defined as
+:ref:`"soft" dependencies <debops_policy__ref_code_standards_soft_role_dependencies>`
+via playbook unless the tight coupling to another role is unavoidable for
+implementing the required functionality.
```

Comment:
This is certainly the case with the `debops.secret` role, where the `secret` variable is directly used in other roles in the `lookup("password")` statements. Therefore if the roles directly use variables from other roles, these roles must be hard dependencies.